File: php/rooms/options/RoomWithChest.php
<?php
namespace rooms\options;

use logic\GameMaster;
use monsters\Hero;
use chest\Chest;

class RoomWithChest extends RoomDecorator{
    private $chest;

    public function __construct(GameMasterable $explorable, Chest $chest) {
        parent::__construct($explorable);
        $this->chest = $chest;
    }

    public function explore(Hero $hero) {
        if($this->chest != null) {
            $this->room->getMuster()->lootChest($hero, $this->chest);
        }
        $this->chest = null;
    }

    public function getNextRooms() {
        return $this->room->getNextRooms();
    }

    public function getRoomType() {
        return $this->room->getRoomType();
    }

    public function getMuster() {
        return $this->room->getMuster();
    }

}
?>

File: java/rooms/options/RoomWithChest.java
package rooms.options;

import logic.GameMaster;
import monsters.Hero;
import chest.Chest;

public class RoomWithChest extends RoomDecorator{
    private Chest chest;

    public RoomWithChest(GameMasterable explorable, Chest chest) {
        super(explorable);
        this.chest = chest;
    }

    @Override
    public void explore(Hero hero) {
        if(chest != null) room.getMuster().lootChest(hero,chest);
        chest = null;
    }

    @Override
    public GameMasterable[] getNextRooms() {
        return room.getNextRooms();
    }

    @Override
    public RoomType getRoomType() {
        return room.getRoomType();
    }

    @Override
    public GameMaster getMuster() {
        return room.getMuster();
    }

}
